src/test/unit_tests/test_lifecycle.js

```diff
@@ -0,0 +1,222 @@
+        });
+        coretest.log('S3 CONFIG', s3.config);
+    });
+
+    mocha.describe('bucket-lifecycle-data-representation', function() {
+        this.timeout(60000);
+
+        mocha.it('test rules length', async () => {
+            await commonTests.test_rules_length(Bucket, Key, s3);
+        });
+        mocha.it('test rule status', async () => {
+            await commonTests.test_rule_status(Bucket, Key, s3);
+        });
+        mocha.it('test expiration date', async () => {
+            await commonTests.test_expiration_date(Bucket, Key, s3);
+        });
+        mocha.it('test rule filter', async () => {
+            await commonTests.test_rule_filter(Bucket, Key, s3);
+        });
+        mocha.it('test expiration days', async () => {
+            await commonTests.test_expiration_days(Bucket, Key, s3);
+        });
+        mocha.it('test filter tag', async () => {
+            await commonTests.test_filter_tag(Bucket, TagName, TagValue, s3);
+        });
+        mocha.it('test and tag', async () => {
+            await commonTests.test_and_tag(Bucket, TagName, TagValue, TagName2, TagValue2, s3);
+        });
+        mocha.it('test and tags prefix days', async () => {
+            await commonTests.test_and_tag_prefix(Bucket, Key, TagName, TagValue, TagName2, TagValue2, s3);
+        });
+        mocha.it('test rule id', async () => {
+            await commonTests.test_rule_id(Bucket, Key, s3);
+        });
+        mocha.it('test empty rule', async () => {
+            await commonTests.test_empty_filter(Bucket, s3);
+        });
+        mocha.it('test rule size', async () => {
+            await commonTests.test_filter_size(Bucket, s3);
+        });
+        mocha.it('test and prefix size', async () => {
+            await commonTests.test_and_prefix_size(Bucket, Key, s3);
+        });
+    });
+
+    mocha.describe('bucket-lifecycle-bg-worker', function() {
+        this.timeout(60000);
+
+        async function create_mock_object(key, bucket, age, size, tagging) {
+            const content_type = 'application/octet-stream';
+            console.log('create_object_upload bucket', bucket, 'key', key, 'content-type', content_type);
+            const { obj_id } = await rpc_client.object.create_object_upload({ bucket, key, content_type });
+            console.log('create_object_upload obj_id', obj_id);
+            const completeUploadResult = await rpc_client.object.complete_object_upload({ obj_id, bucket, key });
+            console.log('completeUploadResult', completeUploadResult);
+
+            // go back in time
+            let create_time = new Date();
+            create_time.setDate(create_time.getDate() - age);
+            const update = {
+                create_time,
+            };
+            if (size) update.size = size;
+            if (tagging) update.tagging = tagging;
+
+            const putObjectPropertyResult = await rpc_client.object.put_object_property({ bucket, key, update });
```

Comment:
I don't think we should add RPC functions just for testing purposes. 
any reason to not call md_store directly? 

```suggestion
            await MDStore.instance().update_object_by_id(obj_id, update);
```

Reply:
👍

Reply:
Tried this one. This code causes a mongo specific RPC error throwing in mongo_client.js:

```
     Error: NO_SUCH_OBJECT
      at MongoClient.check_update_one (src/util/mongo_client.js:336:19)
      at MDStore.update_object_by_id (src/server/object_services/md_store.js:112:30)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async create_mock_object (src/test/unit_tests/test_lifecycle.js:108:40)
      at async Context.<anonymous> (src/test/unit_tests/test_lifecycle.js:173:13)
```

Reply:
this is not mongo specific, it is coming from our code. 
in `update_object_by_id` we call `check_update_one`, which is different in mongo_client vs postgres_client. maybe there is something wrong with the obj id you pass. let me know if you need help debugging it.

Reply:
I am looking into this issue. 


- Ran the test with postgres and all tests passed  - [postgres_lifecycle_index.log](https://github.com/noobaa/noobaa-core/files/8459406/postgres_lifecycle_index.log)
- Ran the test with mongo and some tests failed -
[mongo_lifecycle_index.log](https://github.com/noobaa/noobaa-core/files/8459402/mongo_lifecycle_index.log)

Reply:
@baum, the only thing the RPC adds is a server/client layer, which makes me suspect there is a timing issue here. We can test this theory by adding a small delay before `update_object_by_id`.
If the timing is indeed the issue, then It's worth examining if it is an issue in our code or something in the mongo stack. You can add log messages in the md_store right before calling the DB functions, to make sure they are executed in the correct order.